src/btree/row_srch.c

```diff
@@ -142,6 +149,62 @@ __wt_search_insert(
     cbt->compare = -cmp;
     cbt->ins = (ins != NULL) ? ins : last_ins;
     cbt->ins_head = ins_head;
+
+    /*
+     * FIXME WT-10439 - Available categories are changing in WT-10439. Whichever change merges
+     * second needs to change this to WT_DIAGNOSTIC_BTREE_VALIDATE.
+     */
+    if (EXTRA_DIAGNOSTICS_ENABLED(session, WT_DIAG_DATA_VALIDATION))
+        WT_RET(__validate_next_stack(session, cbt->next_stack));
```

Comment:
This will always run in diagnostic mode. I'm a bit surprised, but it doesn't seem to hurt performance all that much even though we're effectively doubling the number of key comparisons per search_insert call. I want to double check this before opening the PR for review